src/main/java/com/aws/greengrass/lifecyclemanager/KernelLifecycle.java

```diff
@@ -457,11 +457,24 @@ public void startupAllServices() {
      * Shutdown all services in dependency order.
      *
      * @param timeoutSeconds timeout seconds for waiting all services to shutdown. Use -1 to wait infinitely.
+     * @param serviceNames Configuration of services yet to be merged
      */
     @SuppressWarnings("PMD.AvoidCatchingThrowable")
-    public void stopAllServices(int timeoutSeconds) {
-        GreengrassService[] d = kernel.orderedDependencies().toArray(new GreengrassService[0]);
+    public void stopAllServices(int timeoutSeconds, Set<String> serviceNames) {
+        HashSet<String> runningServices = kernel.orderedDependencies().stream().map(GreengrassService::getName)
+                .collect(Collectors.toCollection(HashSet::new));
+        ArrayList<GreengrassService> newServices = serviceNames.stream()
+                .filter(service -> !runningServices.contains(service))
+                .collect(Collectors.toCollection(ArrayList::new))
+                .stream()
+                .map(kernel::locateIgnoreError)
```

Comment:
nit: why do you even need to filter for "running services"
couldn't you just do something like

```
.map(kernel::locateIgnoreError)
.filter(s -> s.getClass().instanceOf(UnloadableService.class))
.collect(...)
```

you shouldn't have to collect and then stream either, you can chain the filter and map together